### PR TITLE
Auto accept chat join requests (tested)

### PR DESCRIPTION
**Pull request by si-nn-er-ls**

### CODE REVIEW

**Comment by si-nn-er-ls** [comment]
@SnousBot

**Review by TheHamkerCat — CHANGES_REQUESTED** [review]
Use [wbb/core/decorators/permissions.py#L63C5-L63C15 (adminsOnly decorator)](https://github.com/TheHamkerCat/WilliamButcherBot/blob/dev/wbb/core/decorators/permissions.py#L63C5-L63C15)

For permissions

**Comment by si-nn-er-ls** [comment]
Updated as you asked

**Comment by TheHamkerCat** [comment]
Why do you still have this call?

https://github.com/TheHamkerCat/WilliamButcherBot/pull/220/commits/a7cf7e009d59f92786e8cc951d1f6ad376a79ff5#diff-ee3116da6c0e3281f6a96cc5431fd96a8d569f102885181c618eac8c621a2829R25

**Comment by si-nn-er-ls** [comment]
My bad. I did it In a hurry 🙃

**Comment by si-nn-er-ls** [comment]
I think it's done for now

File: wbb/modules/greetings.py [hunk 1]
```diff
@@ -107,112 +107,119 @@ async def get_initial_captcha_cache():
...
+    await update_captcha_cache(answers_dicc)
+
+    asyncio.create_task(
+        kick_restricted_after_delay(WELCOME_DELAY_KICK_SEC, button_message, member)
+    )
+    await asyncio.sleep(0.5)
 
+@app.on_message(filters.new_chat_members, group=welcome_captcha_group)
+@capture_err
+async def welcome(_, message: Message):
```
Comment (TheHamkerCat) [inline wbb/modules/greetings.py:218]: this handler (with filters.new_chat_members) won't be called by pyrogram when the chat join request gets auto approved?

Did you properly check if handle_new_member() is not being called twice? (by welcome() and the manual call  in autoapprove.py at the end)

Comment (si-nn-er-ls) [reply wbb/modules/greetings.py:218]: Nope it won't. If you want you can verify https://t.me/+PR9SmmwO5GkwZDg1

**Review by TheHamkerCat — APPROVED** [review]

**Pull request merged**